### PR TITLE
RFC 0011: Allow Intrinsic Functions and Pseudo-Parameter References in DeletionPolicy and UpdateReplacePolicy RFC

### PR DESCRIPTION
**Pull request by MalikAtalla-AWS**
### Language Enhancement Request For Comment

This is a request for comments about allowing Intrinsic Functions and Pseudo-Parameter References in `DeletionPolicy` and  `UpdateReplacePolicy` Resource Attributes RFC. See the tracking issue (https://github.com/aws-cloudformation/cfn-language-discussion/issues/11) for additional details. 

---

## Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jlhood — APPROVED** [review]

File: RFCs/0011-DeletionPolicy.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    * [Fn::FindInMap](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-findinmap.html)
+    * [Fn::If](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-conditions.html#intrinsic-function-reference-conditions-if)
+
+
+* The following Pseudo-Parameters will be supported
+    * `AWS::AccountId`, `AWS::Region`, `AWS::Partition`
+
+# Limitation
+
+Does not support expressions containing references to *[Resource properties](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-template-resource-type-ref.html)* or *certain pseudo parameters*
```
Comment (benkehoe) [inline RFCs/0011-DeletionPolicy.md:154]: Why does it have this limitation?

Comment (MalikAtalla-AWS) [reply RFCs/0011-DeletionPolicy.md:156]: Good question. These are short-term limitations due to underlying implementation constraints. In the fullness of time, these limitations should be removed.

**Review by lejiati — APPROVED** [review]

**Review by jlhood — DISMISSED** [review]

File: RFCs/0011-DeletionPolicy.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+    * [Fn::FindInMap](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-findinmap.html)
+    * [Fn::If](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-conditions.html#intrinsic-function-reference-conditions-if)
+
+
+* The following Pseudo-Parameters will be supported
+    * `AWS::AccountId`, `AWS::Region`, `AWS::Partition`
+
+# Limitation
+
+**Note:** These are short-term limitations due to underlying implementation constraints. In the fullness of time, these limitations should be removed.
```
Comment (benbridts) [inline RFCs/0011-DeletionPolicy.md:154]: I would remove "short term" in this description.

Comment (MalikAtalla-AWS) [reply RFCs/0011-DeletionPolicy.md]: Thanks. Yeah, makes sense to keep it consistent with the other PRs.

**Review by lejiati — APPROVED** [review]

**Review by jlhood — APPROVED** [review]

**Pull request merged**